core/src/main/java/jenkins/FilePathFilterAggregator.java

```diff
@@ -31,9 +31,10 @@ private Entry(FilePathFilter filter, double ordinal) {
 
         @Override
         public int compareTo(Entry that) {
-            double d = this.ordinal - that.ordinal;
-            if (d<0)    return -1;
-            if (d>0)    return 1;
+            double result = Double.compare(this.ordinal, that.ordinal);
+
+            if (result < 0) return -1;
+            if (result > 0) return 1;
```

Comment:
Nit: I think you can simplify this to just `return Double.compare(this.ordinal, that.ordinal);`

Reply:
According to the docs it says greater / less than zero depending on the result but not specifically 1 / -1 so that possibly changes behaviour

Reply:
True, though comparators are supposed to simply check for greater/less than zero. No need to change the behavior here just in case, though.